### PR TITLE
TASS-450: Add module buildkite_elastic_stack_v5 using the latest stack

### PR DESCRIPTION
**Pull request by jmks**
Updates module `buildkite_elastic_stack` using Buildkite's newest CI stack: https://github.com/buildkite/elastic-ci-stack-for-aws

### CODE REVIEW

File: modules/buildkite_elastic_stack_v5/variables.tf [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  default = ""
+}
+
+variable "buildkite_agent_token" {
+}
+
+variable "buildkite_queue" {
+}
+
+variable "enable_experimental_lambda_based_autoscaling" {
```
Comment (jmks) [inline modules/buildkite_elastic_stack_v5/variables.tf:19]: Delete this

File: modules/buildkite_elastic_stack_v5/variables.tf [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  description = "Uses a custom lambda for autoscaling vs the standard AWS AutoScaling"
+  default     = false
+}
+
+variable "instance_type" {
```
Comment (jmks) [inline modules/buildkite_elastic_stack_v5/variables.tf:24]: Add default. Give suggestions for test runners, builders and deployers.

File: modules/buildkite_elastic_stack_v5/variables.tf [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+variable "max_size" {
+}
+
+variable "min_size" {
+  default = 0
+}
+
+variable "root_volume_size" {
+  description = "Size of each instance's root EBS volume (in GB)"
+  default     = 250
```
Comment (jmks) [inline modules/buildkite_elastic_stack_v5/variables.tf:43]: This seems quite large?
Is it supposed to scale with instance size?

File: modules/buildkite_elastic_stack_v5/variables.tf [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+variable "scale_adjustment" {
```
Comment (jmks) [inline modules/buildkite_elastic_stack_v5/variables.tf:46]: Add definition from stack.yml

File: modules/buildkite_elastic_stack_v5/variables.tf [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  default = 0
+}
+
+variable "scale_in_idle_period" {
```
Comment (jmks) [inline modules/buildkite_elastic_stack_v5/variables.tf:50]: Add definition from stack.yml

File: modules/buildkite_elastic_stack_v5/variables.tf [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+variable "spot_price" {
+  default = 0
+}
+
+variable "stack_name" {
+}
+
+variable "stack_template_url" {
```
Comment (jmks) [inline modules/buildkite_elastic_stack_v5/variables.tf:64]: This should depend on the version of the stack you're using
Maybe that's how we specify the version...🤔

Comment (jmks) [reply modules/buildkite_elastic_stack_v5/variables.tf:64]: Yes, we parameterize the stack version into the URL to grab (from buildkite's public s3 bucket)

File: modules/buildkite_elastic_stack_v5/variables.tf (hunk 2 above)
Comment (jmks) [reply modules/buildkite_elastic_stack_v5/variables.tf:24]: They mostly have defaults elsewhere

**Review by cameronr-nulogy — APPROVED** [review]

**Pull request merged**